app/helpers/application_helper.rb

```diff
@@ -1,2 +1,11 @@
 module ApplicationHelper
+  # Returns the full title on a per-page basis.
+  def full_title page_title = ""
+    base_title = t ".title"
+    if page_title.empty?
```

Comment:
phân biệt cho a empty? blank? present? và nil? nhé..

Reply:
-.nil? 	
	+Tất cả đối tượng đều respond
	+Trả về true khi biến được cấp phát bộ nhớ nhưng chưa được gán giá trị
-.empty?
	+Chỉ respond với các instance String, Array, Hash, Set
	+Trả về true khi length = 0
-.blank? 
	+Chỉ có trong Rails
	+Trả về true khi đối tượng là
		#instance of NilClass
		#instance of FalseClass
		#string chỉ chứa khoảng trống
		#array có 0 phần tử
		#hash có 0 phần tử
		#set có 0 phần tử
	+Trả về false khi đối tượng không thuộc các đối tượng trả về true
-.present?
	+Chỉ có trong Rails
	+giá trị trả về là phủ định của .blank?